{"version":3,"sources":["../../app-src/util/ApplicationException.js"],"names":["ApplicationException","Error","constructor","msg","name","exception","isApplicationException","err","Object","getPrototypeOf","getExceptionMessage","message","console","log"],"mappings":";;;;;;AAAO,kBAAMA,oBAAN,SAAmCC,KAAnC,CAAyC;AAC5CC,4BAAYC,MAAM,EAAlB,EAAsB;AAClB,0BAAMA,GAAN;AACA,yBAAKC,IAAL,GAAY,KAAKF,WAAL,CAAiBE,IAA7B;AACH;AAJ2C;;;;AAOhD,kBAAMC,YAAYL,oBAAlB;AACO,qBAASM,sBAAT,CAAgCC,GAAhC,EAAqC;AACxC,uBAAOA,eAAeF,SAAf,IACHG,OAAOC,cAAP,CAAsBF,GAAtB,aAAsCF,SAD1C;AAEH;;;;AAEM,qBAASK,mBAAT,CAA6BH,GAA7B,EAAkC;AACrC,oBAAID,uBAAuBC,GAAvB,CAAJ,EAAiC;AAC7B,2BAAOA,IAAII,OAAX;AACH,iBAFD,MAEO;AACHC,4BAAQC,GAAR,CAAYN,GAAZ;AACA,2BAAO,uCAAP;AACH;AACJ","file":"ApplicationException.js","sourcesContent":["export class ApplicationException extends Error {\n    constructor(msg = '') {\n        super(msg);\n        this.name = this.constructor.name;\n    }\n}\n\nconst exception = ApplicationException;\nexport function isApplicationException(err) {\n    return err instanceof exception ||\n        Object.getPrototypeOf(err) instanceof exception\n}\n\nexport function getExceptionMessage(err) {\n    if (isApplicationException(err)) {\n        return err.message;\n    } else {\n        console.log(err);\n        return 'Não\tfoi\tpossível\trealizar\ta\toperação.';\n    }\n}"]}